lib/ecto/query/planner.ex

```diff
@@ -116,6 +116,12 @@ defmodule Ecto.Query.Planner do
   The cache value is the compiled query by the adapter
   along-side the select expression.
   """
+  def query({%Ecto.Query{} = query, _prefix, _table, _on_conflict, _return_sources} = tuple, :insert_all, cache, adapter, counter) do
```

Comment:
Instead of passing the whole tuple here, pass only the query and handle the tuple upstream. :)